crates/binjs_io/src/entropy/write/lazy_stream.rs

```diff
@@ -24,14 +25,29 @@ const BROTLI_LG_WINDOW_SIZE: u32 = 20;
 /// all data to an external file, for
 /// forensics purposes.
 
+struct Dumps {
+    raw: std::fs::File,
+    brotli: brotli::CompressorWriter<std::fs::File>,
+}
+impl Write for Dumps {
+    fn write(&mut self, data: &[u8]) -> Result<usize, std::io::Error> {
```

Comment:
Suggestion: maybe provide this as an implementation for `write_all` instead (which this essentially is), and then let `write` call into that but return `Ok(data.len())`.